src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultCollectionValidationStrategy.cs

```diff
@@ -45,6 +46,7 @@ public class DefaultCollectionValidationStrategy : IValidationStrategy
         /// Gets an instance of <see cref="DefaultCollectionValidationStrategy"/>.
         /// </summary>
         public static readonly IValidationStrategy Instance = new DefaultCollectionValidationStrategy();
+        private static readonly Dictionary<Type, MethodInfo> _genericGetEnumeratorCache = new Dictionary<Type, MethodInfo>();
```

Comment:
Just make `GetEnumeratorForElementType()` non-`static`. This is an `.Internal` class and such minor changes are fair game.